src/scss/components/_btn_text.scss

```diff
@@ -0,0 +1,131 @@
+// ratio: 3.0, 4.5 (= AA compliant), or 7.0
+@function a11y-color($color-1, $color-2, $lod-color-2: 'null', $ratio: 4.5) {
+  $adjust-color-1: 'false';
+  // Calculate the first contrast ratio of the given pair
+  $original-contrast: contrast-ratio($color-1, $color-2);
+
+  @if $original-contrast >= $ratio {
+    // If we pass the ratio already, return the original color
+    @return ($color-1, $color-2);
+  } @else {
+    // Doesn't pass
+    // Should the color be lightened or darkened?
+    $color-1-lod: light-or-dark($color-1);
+    $color-2-lod: light-or-dark($color-2);
+
+    // Set a "step" value to lighten or darken a color
+    // Note: Higher percentage steps means faster compile time, but we might overstep the required threshold too far with something higher than 5%
+    $step: 2%;
+
+    @if $color-2-lod == 'light' and $lod-color-2 == 'lighter' {
+      // if $color-1 is light and $lod-color-2 is lighter -> darken $color-2
+      $adjust-color-1: 'true';
+      $step: - $step;
+    }
+    @else if $color-2-lod == 'light' and $color-1-lod == 'light' {
+      // Both are light colors, darken the color-1
+      $step: - $step;
+    } @else if $color-2-lod == 'dark' and $color-1-lod == 'light' {
+      // color-1 is light, color-2 is dark but does not pass, darken more
+      $step: - $step;
+    }
+
+    // Loop through with a @while statement until the color combination passes our required ratio. Scale the color by our step value until the expression is false
+    // This might take some depending on the colors -> pay attention that the loop is not infinite!
+    @while contrast-ratio($color-1, $color-2) < $ratio {
+      @if $adjust-color-1 == 'true' {
+        $color-1: scale-color($color-1, $lightness: $step, $saturation: $step/2)
+      }
+      @else {
+        $color-2: scale-color($color-2, $lightness: $step, $saturation: $step/2);
+      }
+    }
+    @return ($color-1, $color-2);
+  }
+}
+
+
+@mixin btn-text-common() {
+
+  // Align all children vertically
+  > * {
+    vertical-align: middle;
+  }
+}
+
+
+@each $color, $value in $theme-colors {
+  $colors: a11y-color(if($color=='dark', tint-color($value, 10%), shade-color($value, 10%)), if($color=='dark', tint-color($value, 90%), tint-color($value, 85%)), 'lighter');
+  $fg: nth($colors, 1);
+  $bg: nth($colors, 2);
+  $colors-active: a11y-color(shade-color($value, 20%), tint-color($value, 65%), 'lighter');
+  $fg-active: nth($colors-active, 1);
+  $bg-active: nth($colors-active, 2);
+
+  // btn-text has no background or border, and only sets the font-color.
+  .btn-text-#{$color} {
+    @include button-variant(
+      transparent, // background
+      transparent, // border
+      if($color=='dark', tint-color($value, 10%), $value), // color
+
+      transparent, // hover-background
+      transparent, // hover-border
+      if($color=='dark', tint-color($value, 25%), tint-color($value, 20%)), // hover-color
```

Comment:
@dvdanielamoitzi @thinkh Are we sure we want to have this logic just for a 5% difference in tint? I am not sure if that is even noticeable. 
Generally: we are only checking for `dark`, but what if I add a custom theme color with the same color-value as `dark`? It will not apply all those slight changes then I guess. All these ifs seem very brittle and hand-tailored for our custom theme colors, I think it is worth investing time in finding a logic that works for all color values, i.e. by using the light-or-dark? It has to be discussed if we can find a solution that simply uses the color-value, not the name.
Imagine if I set `dark` to `#ffffff` (or a not so dark dark like right now), that is perfectly possible as the semantics of the name do not infer a color-value in all cases.

Reply:
That is also why Bootstrap (https://getbootstrap.com/docs/5.1/components/buttons/#mixins) uses the if to determine the light/dark status, and then darkens/lightens accordingly. 
So in words: 
If color **value** is dark (as most colors except light): background should be lightened version. 
If color **value** is light: background should be darkened version. 
Same with hover and active (should make dark darker, and light lighter). 
Do you know what I mean with that?

Reply:
Yes, I can follow your arguments. I think, we should create a CodePen with Bootstrap and Fontawesome included and add all possible buttons. Then we can make a more profound decission. What do you think?

Reply:
👍 
I would also be interested in how the buttons (maybe a light variant) behaves on a `bg-dark` for example.